nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ExtractEmailHeaders.java

```diff
@@ -0,0 +1,234 @@
+                    " Values will respect email email contents.")
+            .required(false)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .defaultValue("x-mailer")
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Extraction was successful")
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Flowfiles that could not be parsed as a RFC-2822 compliant message")
+            .build();
+
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> descriptors;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+
+        descriptors.add(CAPTURED_HEADERS);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        final ComponentLog logger = getLogger();
+        final Map<String, String> attributes = new HashMap<>();
+        final FlowFile originalFlowFile = session.get();
+        if (originalFlowFile == null) {
+            return;
+        }
+
+
+        final List<String> capturedHeadersList = Arrays.asList(context.getProperty(CAPTURED_HEADERS).getValue().toLowerCase().split(":"));
+
+
+        session.read(originalFlowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream rawIn) throws IOException {
+                    try (final InputStream in = new BufferedInputStream(rawIn)) {
+                        Properties props = new Properties();
+                        Session mailSession = Session.getDefaultInstance(props, null);
+                        MimeMessage originalMessage = new MimeMessage(mailSession, in);
+                        MimeMessageParser parser = new MimeMessageParser(originalMessage).parse();
+                        // RFC-2822 determines that a message must have a "From:" header
+                        // if a message lacks the field, it is flagged as invalid
+                        Address[] from = originalMessage.getFrom();
+                        Date sentDate = originalMessage.getSentDate();
+                        if (from == null || sentDate == null ) {
+                            // See ya later my friend...
+                            throw new MessagingException("Message failed RFC2822 validation");
```

Comment:
I see...

 I changed the processor a bit so that it follows the same logic as the `ExtractEmailAttachments` (i.e. if message fails parsing, `MessagingExceptionis `raised and captured via `catch `and added to `invalidFlowFileList`. Code continues until `session.transfer(invalidFlowFilesList, REL_FAILURE);` )

Let me know if the change is enough